src/zarr/core/array.py

```diff
@@ -1130,7 +1131,61 @@ async def _delete_key(key: str) -> None:
 
         # Write new metadata
         await self._save_metadata(new_metadata)
-        return replace(self, metadata=new_metadata)
+
+        # Update metadata (in place)
+        object.__setattr__(self, "metadata", new_metadata)
```

Comment:
@TomAugspurger - calling this out because it will impact the xarray use of resize... now in-place again :)